src/OrchardCore/OrchardCore.Data/SessionHelper.cs

```diff
@@ -39,22 +39,23 @@ public async Task<T> LoadForUpdateAsync<T>(Func<T> factory = null) where T : cla
         }
 
         /// <inheritdocs/>
-        public async Task<T> GetForCachingAsync<T>(Func<T> factory = null) where T : class, new()
+        public async Task<(bool, T)> GetForCachingAsync<T>(Func<T> factory = null) where T : class, new()
```

Comment:
More like `GetOrCreate` than `Get`. That would make the factory usage obvious.

Reply:
So here, in the `distributedCache` branch, it is done at the `IDocumentStore` (can be an `IFileDocumentStore`) level, same problem but cacheable is only checked at the `IDocumentManager` level.

> More like GetOrCreate than Get. That would make the factory usage obvious.

Yes i agree, before the change it was already using a factory, would you mean `GetOrCreateForCachingAsync`? In the `distributedCache` branch the related method is `GetImmutableAsync()` that would become `GetOrCreateImmutableAsync()`

I will think about it, but again i'm open to any suggestions, let me know